x/cronos/keeper/precompiles/relayer.go

```diff
@@ -119,37 +89,37 @@ func (bc *RelayerContract) Run(evm *vm.EVM, contract *vm.Contract, readonly bool
 	)
 	switch prefix {
 	case prefixCreateClient:
-		res, err = exec(bc, stateDB, input, new(clienttypes.MsgCreateClient), bc.ibcKeeper.CreateClient)
+		res, err = exec(bc.cdc, stateDB, contract.CallerAddress, input, bc.ibcKeeper.CreateClient)
```

Comment:
do we need to check evm.Origin as well?

Reply:
caller will allow contract to manage it's own stuff.